From deb7a5aa682cc4143fc3a5bd2d453b1a688fe16c Mon Sep 17 00:00:00 2001
From: balajhidn <balajhix.neelakantan.durvas@intel.com>
Date: Thu, 2 Mar 2023 18:05:09 +0000
Subject: [PATCH] Fix Crashlog collect task status

Updated interface path

Tested:
Triggered crashlog with post request from redfish
Check for response is 202 Accepted
TaskState shows Completed and crashlog is generated

Signed-off-by: Gade-Rajasekhar Reddy <raja.sekhar.reddy.gade@intel.com>
Signed-off-by: balajhidn <balajhix.neelakantan.durvas@intel.com>
Upstream-Status: Pending
---
 redfish-core/lib/log_services.hpp | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/redfish-core/lib/log_services.hpp b/redfish-core/lib/log_services.hpp
index 8267cd0f..23c40b67 100644
--- a/redfish-core/lib/log_services.hpp
+++ b/redfish-core/lib/log_services.hpp
@@ -4976,10 +4976,11 @@ inline void requestRoutesCrashlogCollect(App& app)
         {
             iface = crashlogSystemDiagnosticsInterface;
             method = "CreateDump";
-            taskMatchStr = "type='signal',"
-                           "interface='org.freedesktop.DBus.ObjectManager',"
-                           "member='InterfacesAdded',"
-                           "path='xyz/openbmc_project/dump/crashlog'";
+            taskMatchStr =
+                "type='signal',"
+                "interface='org.freedesktop.DBus.ObjectManager',"
+                "member='InterfacesAdded',"
+                "path_namespace='/xyz/openbmc_project/dump/crashlog'";
         }
         else
         {
-- 
2.25.1

